server/chat-plugins/datasearch.ts

```diff
@@ -1104,8 +1104,23 @@ function runDexsearch(target: string, cmd: string, canAll: boolean, message: str
 		Object.values(search).reduce(accumulateKeyCount, 0)
 	));
 
+	// Prepare move validator and pokemonSource outside the hot loop
+	// but don't prepare them at all if there are no moves to check...
+	// These only ever get accessed if there are moves to filter by.
+	let validator;
+	let pokemonSource;
+	if (Object.values(searches).some(search => Object.keys(search.moves).length !== 0)) {
+		const format = Object.entries(Dex.data.Rulesets).find(([a, f]) => f.mod === usedMod)?.[1].name || 'gen9ou';
```

Comment:
I'd prefer not to hardcode format IDs (we inevitably forget when a new generation comes out). If one does need hardcoded, I think gen{whatevergenislatest}anythinggoes would be preferable to check against instead of OU in principle, in case OU for some reason ever modifies the validator.

Reply:
Never mind - this was just moved outside the loop.